src/simularium/VisDataCache.ts

```diff
@@ -36,14 +39,44 @@ class VisDataCache {
     public changeSettings(options: {
         maxSize?: number;
         cacheEnabled?: boolean;
+        onUpdate?: (log: CacheLog) => void;
     }): void {
-        const { maxSize, cacheEnabled } = options;
+        const { maxSize, cacheEnabled, onUpdate } = options;
         if (cacheEnabled !== undefined) {
             this._cacheEnabled = cacheEnabled;
         }
         if (maxSize !== undefined) {
             this._maxSize = maxSize;
         }
+        if (onUpdate !== undefined) {
+            this.cacheUpdateCallback = onUpdate;
+        }
+    }
+
+    public onCacheUpdate(): void {
+        if (this.cacheUpdateCallback) {
+            this.cacheUpdateCallback({
+                size: this.size,
+                numFrames: this.numFrames,
+                maxSize: this._maxSize,
+                enabled: this._cacheEnabled,
+                firstFrameNumber: this.getFirstFrameNumber(),
+                firstFrameTime: this.getFirstFrameTime(),
+                lastFrameNumber: this.getLastFrameNumber(),
+                lastFrameTime: this.getLastFrameTime(),
+                framesInCache: this.getListOfCachedFrameNumbers(),
```

Comment:
feedback for future work: separate out things that we truly need for the installing app to know about, and what you're using for debugging. either in the Advanced settings panel, or a dev mode check or something.